numba_dppy/ocl/ocldecl.py

```diff
@@ -35,66 +35,32 @@
 # register_number_classes(intrinsic_global)
 
 
-@intrinsic
-class Ocl_get_global_id(ConcreteTemplate):
-    key = dppy.get_global_id
-    cases = [signature(types.intp, types.uint32)]
-
+def register(function, args):
+    base = ConcreteTemplate
+    name = f"Ocl_{function.__name__}"
 
-@intrinsic
-class Ocl_get_local_id(ConcreteTemplate):
-    key = dppy.get_local_id
-    cases = [signature(types.intp, types.uint32)]
+    if isinstance(args, list):
+        cases = [signature(*a) for a in args]
+    else:
+        cases = [signature(*args)]
 
+    dct = {"key": function, "cases": cases}
+    cls = type(base)(name, (base,), dct)
 
-@intrinsic
-class Ocl_get_group_id(ConcreteTemplate):
-    key = dppy.get_group_id
-    cases = [signature(types.intp, types.uint32)]
-
-
-@intrinsic
-class Ocl_get_num_groups(ConcreteTemplate):
-    key = dppy.get_num_groups
-    cases = [signature(types.intp, types.uint32)]
-
-
-@intrinsic
-class Ocl_get_work_dim(ConcreteTemplate):
-    key = dppy.get_work_dim
-    cases = [signature(types.uint32)]
-
-
-@intrinsic
-class Ocl_get_global_size(ConcreteTemplate):
-    key = dppy.get_global_size
-    cases = [signature(types.intp, types.uint32)]
-
-
-@intrinsic
-class Ocl_get_local_size(ConcreteTemplate):
-    key = dppy.get_local_size
-    cases = [signature(types.intp, types.uint32)]
-
-
-@intrinsic
-class Ocl_barrier(ConcreteTemplate):
-    key = dppy.barrier
-    cases = [signature(types.void, types.uint32), signature(types.void)]
-
-
-@intrinsic
-class Ocl_mem_fence(ConcreteTemplate):
-    key = dppy.mem_fence
-    cases = [signature(types.void, types.uint32)]
-
-
-@intrinsic
-class Ocl_sub_group_barrier(ConcreteTemplate):
-    key = dppy.sub_group_barrier
+    globals()[name] = intrinsic(cls)
+    intrinsic_global(function, types.Function(cls))
```

Comment:
The only new thing is here. 
Other changes are only for refactoring and testing.